test/unit/lib/runner/writers/action-json-writer-tests.js

```diff
@@ -0,0 +1,170 @@
+'use strict';
+
+const mockery = require('mockery');
+const sinon = require('sinon');
+const expect = require('chai').expect;
+
+describe('lib/runner/test-runner/action-json-writer.js', function() {
+  describe('writeActionReport', function() {
+    let fs;
+    let path;
+    let writeActionReportToDisk;
+    let configHandler;
+
+    let report;
+
+    beforeEach(function() {
+      mockery.enable({useCleanCache: true});
+      mockery.registerAllowable('../../../../../lib/runner/writers/action-json-writer.js');
+
+      path = {
+        resolve: sinon.stub(),
+      };
+
+      report = {
+        testReports: [
+          {
+            testRuns: [
+              {
+                report: {
+                  testName: 'testOne',
+                  actions: [
+                    {
+                      component: 'navigate',
+                      action: 'navigate',
+                    },
+                  ],
+                },
+              },
+              {
+                report: {
+                  testName: 'testOne',
+                  actions: [
+                    {
+                      component: 'modal',
+                      action: 'click',
+                    },
+                  ],
+                },
+              },
+            ],
+          },
+          {
+            testRuns: [
+              {
+                report: {
+                  testName: 'testOne',
+                  actions: [
+                    {
+                      component: 'navigate',
+                      action: 'navigate',
+                    },
+                  ],
+                },
+              },
+              {
+                report: {
+                  testName: 'testOne',
+                  actions: [
+                    {
+                      component: 'modal',
+                      action: 'click',
+                    },
+                  ],
+                },
+              },
+            ],
+          },
+        ],
+      };
+
+      fs = {
+        writeFileSync: sinon.stub(),
+        readFileSync: sinon.stub(),
+      };
+
+      configHandler = {
+        get: sinon.stub(),
+      };
+
+      obj = {};
+
+      mockery.registerMock('fs', fs);
+      mockery.registerMock('path', path);
+      mockery.registerMock('../../util/config-handler.js', configHandler);
+
+      writeActionReportToDisk = require('../../../../../lib/runner/writers/action-json-writer.js');
+    });
+
+    afterEach(function() {
+      mockery.resetCache();
+      mockery.deregisterAll();
+      mockery.disable();
+    });
+
+    it('should call path.resolve with the configs testPath', function() {
+      let report = {
```

Comment:
maybe give either these 'report' varibles you declare specifically a different name than the one you declare in the before each for clarity. or you can remove the 'let' portion, as its already declared and you are reassinging the before each 'report'.  Also it could be you dont need a new report declared here and can just use the one in beforeeach now that you have one there.